lib/src/sql/error/mod.rs

```diff
@@ -169,6 +174,112 @@ impl Location {
 		}
 		unreachable!()
 	}
+
+	#[cfg(feature = "experimental_parser")]
+	pub fn of_span_start(source: &str, span: Span) -> Self {
+		// Bytes of input before substr.
+		let offset = span.offset as usize;
+		// Bytes of input prior to line being iteratated.
+		let mut bytes_prior = 0;
+		for (line_idx, line) in source.split('\n').enumerate() {
```

Comment:
This file seems to be unused I will remove it but this line is copied to the new place in `crate::syn::common`. The reason I split on `\n` is because this function is searching for a byte location within a line. Splitting on lines might eat either one or two bytes without indicating which one. This function would then need to work around that to keep track of how much bytes where eaten.

Reply:
I switched the implementation to an iterator which supports more types of line terminators.